batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/controller/StudentController.java

```diff
@@ -0,0 +1,201 @@
+            List<Student> studentList = studentService.fetch(page);
+            req.setAttribute("studentList", studentList);
+            req.setAttribute("nextPageNum", page+1);
+            req.getServletContext().getRequestDispatcher("/WEB-INF/views/students.jsp").forward(req, resp);
+        } catch (DataException e) {
+            e.printStackTrace();
+            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            req.setAttribute("errorMessage", e.getMessage());
+            req.getServletContext().getRequestDispatcher("/WEB-INF/views/error.jsp").forward(req, resp);
+        }
+    }
+
+
+    private void create(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        req.getServletContext().getRequestDispatcher("/WEB-INF/views/createStudent.jsp").forward(req, resp);
+    }
+
+
+    private void createProcess(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String name= req.getParameter("name");
+        String address = req.getParameter("address");
+        String dob = req.getParameter("dob");
+        String department = req.getParameter("department");
+        String batch = req.getParameter("batch");
+        String rollText = req.getParameter("roll");
+
+        try {
+            DateFormat dateFormat = new SimpleDateFormat("yyyy-mm-dd", Locale.ENGLISH);
+            Date date = dateFormat.parse(dob);
+            Integer roll = Integer.parseInt(rollText);
+
+            Student student = new Student();
+            student.setBatch(batch);
+            student.setDepartment(department);
+            student.setRoll(roll);
+
+            student.setName(name);
+            student.setAddress(address);
+            student.setDob(date);
+
+            try {
+                studentService.insert(student);
+                resp.sendRedirect("/students");
+            } catch (DataException e) {
+                e.printStackTrace();
+                resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                req.setAttribute("errorMessage", e.getMessage());
+                req.getServletContext().getRequestDispatcher("/WEB-INF/views/error.jsp").forward(req, resp);
+            }
+        } catch (ParseException e) {
+            req.setAttribute("errorMessage", "error while parsing date");
+            create(req, resp);
+        } catch (NumberFormatException e){
+            req.setAttribute("errorMessage", "number format error in 'roll'");
+            create(req, resp);
+        }
+    }
+
+
+    private void edit(HttpServletRequest req, HttpServletResponse resp, Integer id) throws ServletException, IOException {
+
+        try {
+            Student student = studentService.fetchById(id);
+            req.setAttribute("student",student);
+            req.getServletContext().getRequestDispatcher("/WEB-INF/views/editStudent.jsp").forward(req, resp);
+        } catch (DataException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)